examples/cloud-datafusion-functions-plugins/src/main/java/com/google/plugin/TruncateTableAction.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.plugin;
+
+import com.google.functions.TruncateTableFunction;
+import io.cdap.cdap.api.TxRunnable;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.DatasetContext;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.action.Action;
+import io.cdap.cdap.etl.api.action.ActionContext;
+
+/** An Action Plugin to truncate the records in a BigQuery table. */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name(TruncateTableAction.NAME)
+@Description("Truncate bigquery table")
+public class TruncateTableAction extends Action {
+  public static final String NAME = "TruncateTableAction";
+  private final Conf config;
+
+  public TruncateTableAction(Conf config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {}
+
+  @Override
+  public void run(ActionContext context) throws Exception {
```

Comment:
What kind of exception can this method throw? Please narrow it down to the specific types.

Reply:
We're just overriding the run method of TxRunnable which throws Exception so there's not much we can do: https://cdap.atlassian.net/wiki/spaces/DOCS/pages/480477416/Workflows

code to TxRunnable here - https://github.com/cdapio/cdap/blob/develop/cdap-api/src/main/java/io/cdap/cdap/api/TxRunnable.java#L41